lib/ex_aws/sqs/client.ex

```diff
@@ -0,0 +1,122 @@
+  configuration chose, such that it does not need passed in with every request
+
+  Usage:
+  ```elixir
+  defmodule MyApp.SQS do
+    use ExAws.SQS.Client, otp_app: :my_otp_app
+  end
+  ```
+
+  In your config
+  ```elixir
+  config :my_otp_app, ExAws,
+    sqs: [], #SQS config goes here
+  ```
+  """
+
+  @type sqs_permission :: 
+    :all | 
+    :receive_message | 
+    :delete_message |
+    :change_message_visibility | 
+    :get_queue_attributes 
+  @type sqs_acl :: [{binary, sqs_permission}]
+  @type sqs_message_attribute_name ::
+    :all |
+    :sender_id | 
+    :sent_timestamp |
+    :approximate_receive_count |
+    :approximate_first_receive_timestamp |
+    :wait_time_seconds |
+    :receive_message_wait_time_seconds 
+  @type sqs_queue_attribute_name ::
+    :all | 
+    :approximate_number_of_messages |
+    :approximate_number_of_messages_not_visible | 
+    :visibility_timeout |
+    :created_timestamp | 
+    :last_modified_timestamp | 
+    :policy  
+  @type sqs_message :: [
+    {:message_body, binary},
+    {:delay_seconds, 0..900 | none},
+    {:attributes, Keyword.t }
+  ]
+  @type queue_attributes :: [
+    {:visibility_timeout, Integer} |
+    {:policy, binary}
+  ]
+ 
+  @doc "Create queue"
+  defcallback create_queue(queue_name :: binary, visibility_timeout :: 0..43200 | none) :: ExAws.Request.response_t
+
+  @doc "Delete a queue"
+  defcallback delete_queue(queue_name :: binary) :: ExAws.Request.response_t
+
+  @doc "Gets attributes of a SQS Queue"
+  defcallback get_queue_attributes(queue_name :: binary, attribute_names :: [sqs_queue_attribute_name] | :all) :: ExAws.Request.response_t
+
+  @doc "Set attributes of a SQS Queue"
+  defcallback set_queue_attributes(queue_name :: binary, attributes :: queue_attributes) :: ExAws.Request.response_t
+  
+  @doc "Retrieces a list of all the SQS Queues"
+  defcallback list_queues() :: ExAws.Request.response_t
+
+  @doc "Retrieves the dead letter source queues for a given SQS Queue"
+  defcallback get_dead_letter_source_queues(queue_name :: binary) :: ExAws.Request.response_t 
+  @doc "Purge all messages in a SQS Queue"
+  defcallback purge_queue(queue_name :: binary) :: ExAws.Request.response_t
+
+  @doc "Add permission"
+  defcallback add_permission(queue_name :: binary, label :: binary, permissions :: sqs_acl) :: ExAws.Request.response_t
+  
+  @doc "Remove permission"
+  defcallback remove_permission(queue_name :: binary, label :: binary) :: ExAws.Request.response_t
+  
+  @doc "Send a message to a SQS Queue"
+  defcallback send_message(queue_name :: binary, message :: sqs_message) :: ExAws.Request.response_t 
+  defcallback send_message(queue_name :: binary, message_body :: binary, delay_seconds :: 0..900 | none) :: ExAws.Request.response_t
+  
+  @doc "Send up to 10 messages to a SQS Queue in a single request"
+  defcallback send_messages(queue_name :: binary, messages :: [sqs_message, ...]) :: ExAws.Request.response_t 
```

Comment:
This should be `send_message_batch` by the way to correspond to SendMessageBatch. Trust me I prefer send_messages, but if we call it that it's no longer clear if it corresponds to AWS SendMessageBatch or if it loops over the messages file and calls `send_message`